### PR TITLE
Fix build and video init issues

### PR DESCRIPTION
**Pull request by mborgerson**
These commits have been lifted from https://github.com/thrimbor/cromwell/tree/build_fix and re-worded by me for clarity. This PR will allow compilation with modern toolchains.

### CODE REVIEW

**Comment by binarymaster** [comment]
You might want to import this commit as well: https://github.com/binarymaster/cromwell/commit/a3c8c193508e99a624be7c2837497c5931ce217f

**Comment by mborgerson** [comment]
@binarymaster Thanks. I've cherry-picked your commit and reworded for brevity

File: scripts/ldscript-crom.ld [hunk 1]
```diff
@@ -38,18 +40,20 @@ SECTIONS {
...
+	/* ram allocations */
+
 	.data (LOW_ROM + SIZEOF(.text) + SIZEOF(.rodata)) : AT( SIZEOF(.text) + SIZEOF(.rodata) ) {
 		_start_data = .;
 		*(.data);
+        *(.data.*);
 		*(.sdata);
 		*(.sdata2);
 		*(.got);
+        *(.got.*);
```
Comment (JayFoxRox) [inline scripts/ldscript-crom.ld]: Indentation differences (other instances, too)

Comment (mborgerson) [reply scripts/ldscript-crom.ld]: Fixed

**Comment by thrimbor** [comment]
Gave it a quick visual inspection and test (booting my old XDSL copy). Even though I was involved in creating these commits, I think it's fine for me to merge given the broken state master is in.

Thanks!

**Pull request merged**